runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -928,9 +893,49 @@ public synchronized CorfuRuntime connect() {
                     .setTimeoutInMinutesForLoading((int) parameters.fastLoaderTimeout.toMinutes());
             fastLoader.loadMaps();
         }
+
+        gcThread.execute(this::runRuntimeGC);
+
         return this;
     }
 
+    /**
+     * Over time garbage is created on the global log and clients clients sync
+     * parts of the log therefore clients can accumulate garbage overtime. This
+     * method runs a garbage collection process on the different client views,
+     * which discards data before the trim mark (i.e. the point that demarcates
+     * the beginning of the active log).
+     */
+    private void runRuntimeGC() {
```

Comment:
Can you extract GC logic? Create a class for that.
What I am worried about - CorfuRuntime comes bigger again. And also runtime shouldn't be in charge of gc logic

Reply:
It is thought. We GC the runtime views. Adding a class just for one small method seems superfluous.

Reply:
Don't agree, CorfuRuntime is gigantic, adding more and more methods in CorfuRuntime that the runtime not in charge of leads us to troubles. 
If you extract Gc part you can add metrics for that, write a unit test, check that gc actually works, you can move a couple of fileds from CorfuRuntime to you gc class. 
It will be much better and have so many methods in corfuruntime

Reply:
Not sure if you've seen the latest update, but there is a test.

Reply:
Say no more fam. Done.